modules/browsemap/src/test/java/com/sygic/modules/browsemap/BrowseMapFragmentViewModelTest.kt

```diff
@@ -0,0 +1,310 @@
+        browseMapFragmentViewModel.poiDetailDataObservable.test().assertValue(testViewObjectData.toPoiDetailData())
+    }
+
+    @Test
+    fun onMapObjectsReceivedSelectionModeFullMapMarkerTest() {
+        val testMarkerData = MarkerData(48.143489, 17.150560, BasicData("Test"))
+        val testMapMarker = MapMarker.from(testMarkerData).build()
+
+        argumentCaptor<PoiDataManager.Callback>().let { callback ->
+            whenever(poiDataManager.getViewObjectData(any(), callback.capture())).then {
+                callback.firstValue.onDataLoaded(testMarkerData)
+            }
+        }
+
+        browseMapFragmentViewModel.mapSelectionMode = MapSelectionMode.FULL
+        browseMapFragmentViewModel.onMapObjectsReceived(listOf(testMapMarker))
+        verify(poiDataManager).getViewObjectData(eq(testMapMarker), any())
+        browseMapFragmentViewModel.poiDetailDataObservable.test().assertValue(testMarkerData.toPoiDetailData())
+    }
+
+    @Test
+    fun customOnMapClickListenerConsumedTest() {
+        val testMarkerData = MarkerData(48.143489, 17.150560, BasicData("Test"))
+        val testMapMarker = MapMarker.from(testMarkerData).build()
+        val onMapClickListener = mock<OnMapClickListener>()
+
+        whenever(onMapClickListener.onMapClick(any())).thenReturn(true)
+        argumentCaptor<PoiDataManager.Callback>().let { callback ->
+            whenever(poiDataManager.getViewObjectData(any(), callback.capture())).then {
+                callback.firstValue.onDataLoaded(testMarkerData)
+            }
+        }
+
+        browseMapFragmentViewModel.onMapClickListener = onMapClickListener
+        browseMapFragmentViewModel.mapSelectionMode = MapSelectionMode.FULL
+        browseMapFragmentViewModel.onMapObjectsReceived(listOf(testMapMarker))
+        verify(extendedMapDataModel, never()).addOnClickMapMarker(any())
+        verify(poiDataManager).getViewObjectData(eq(testMapMarker), any())
+        verify(onMapClickListener).onMapClick(eq(testMarkerData))
+        browseMapFragmentViewModel.poiDetailDataObservable.test().assertNoValue()
+    }
+
+    @Test
+    fun customOnMapClickListenerNotConsumedTest() {
+        val testMarkerData = MarkerData(48.143489, 17.150560, BasicData("Test"))
+        val testMapMarker = MapMarker.from(testMarkerData).build()
+        val onMapClickListener = mock<OnMapClickListener>()
+
+        argumentCaptor<PoiDataManager.Callback>().let { callback ->
+            whenever(poiDataManager.getViewObjectData(any(), callback.capture())).then {
+                callback.firstValue.onDataLoaded(testMarkerData)
+            }
+        }
+
+        browseMapFragmentViewModel.onMapClickListener = onMapClickListener
+        browseMapFragmentViewModel.mapSelectionMode = MapSelectionMode.FULL
+        browseMapFragmentViewModel.onMapObjectsReceived(listOf(testMapMarker))
+        verify(extendedMapDataModel, never()).addOnClickMapMarker(any())
```

Comment:
ked to nehandlujes sam, nema sa prave marker pridat?

Reply:
je tam bug, budu řešit v tasku MS-5646, který mám už v progresu ;)